validate.js

```diff
@@ -0,0 +1,57 @@
+const bb = require('ssb-bendy-butt')
+const bfe = require('ssb-bfe')
+
+exports.validateSingle = function (contentSection) {
+  if (!Array.isArray(contentSection) || contentSection.length !== 2)
+    return new Error(
+      `invalid message: contentSection ${typeof contentSection} with length ${
+        contentSection.length
+      } is incorrect, expected a list of content and contentSignature`
+    )
+
+  const [content, contentSignature] = contentSection
+
+  if (!Array.isArray(content) || content.length < 3 || content.length > 4)
+    return new Error(
+      `invalid message: content ${typeof contentSection} with length ${
+        content.length
+      } is incorrect, expected a list of type, subfeed, metafeed and an optional nonce`
+    )
+
+  // TODO: might need to check if content is encrypted?
```

Comment:
I was thinking about this as well on the first check, at first I thought you would not get them, but then thinking where we need this. See [hydrate](https://github.com/ssb-ngi-pointer/ssb-meta-feeds/blob/5cbf43c4201d6a251832afde31b1d5c0f35abce7/query.js#L72) where encrypted content signals that it is not for us so should be discarded. For `publish` like we talked about I would assume we use this function before encrypting.

Reply:
Maybe be good this as the first check? To make the error better than the first check.

Reply:
Could you describe how the check for encrypted content would be performed? Would you check `typeof` `contentSection` for `string`?

In the `hydrate` example it appears that encrypted content is implicitly ignored because it is not matched by any of the `.filter()` statements. Do I understand that correctly?

Reply:
That is correct, I was thinking we would add the validate check before those filters (line 77) so that we are sure that only valid things are passed further along as that makes checks easier. And yes a typeof check should do the trick.